python_modules/dagster/dagster/_core/definitions/auto_materialize_rule.py

```diff
@@ -215,6 +318,35 @@ def description(self) -> str:
         """
         ...
 
+    def add_evaluation_data_from_previous_tick(
+        self,
+        context: RuleEvaluationContext,
+        asset_partitions_by_evaluation_data: Mapping[
+            Optional[AutoMaterializeRuleEvaluationData], Set[AssetKeyPartitionKey]
+        ],
+        should_use_past_data_fn: Callable[[AssetKeyPartitionKey], bool],
+    ) -> "RuleEvaluationResults":
+        """Combines a given set of evaluation data with evaluation data from the previous tick.
```

Comment:
Maybe add something like "The returned value will include the union of asset partitions that are in `asset_partitions_by_evaluation_data` and asset partitions that the previous tick decided to materialize, minus asset partitions for which `should_use_past_data_fn` evaluates to `True`"